core/mongoinit.go

```diff
@@ -0,0 +1,141 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+type MongoCredentials struct {
+	AdminUser           string `json:"admin,omitempty"`
+	AdminPasswd         string `json:"adminpasswd,omitempty"`
+	MetadataUser        string `json:"metadata,omitempty"`
+	MetadataPasswd      string `json:"metadatapasswd,omitempty"`
+	CoredataUser        string `json:"coredata,omitempty"`
+	CoredataPasswd      string `json:"coredatapasswd,omitempty"`
+	RulesengineUser     string `json:"rulesengine,omitempty"`
+	RulesenginePasswd   string `json:"rulesenginepasswd,omitempty"`
+	NotificationsUser   string `json:"notifications,omitempty"`
+	NotificationsPasswd string `json:"notificationspasswd,omitempty"`
+	SchedulerUser       string `json:"scheduler,omitempty"`
+	SchedulerPasswd     string `json:"schedulerpasswd,omitempty"`
+	LoggingUser         string `json:"logging,omitempty"`
+	LoggingPasswd       string `json:"loggingpasswd,omitempty"`
+}
+
+func credentialInStore(config *tomlConfig, secretBaseURL string, credPath string, c *http.Client) (bool, error) {
+
+	t, err := getSecret(config.SecretService.TokenFolderPath + "/" + config.SecretService.VaultInitParm)
+	if err != nil {
+		lc.Error(err.Error())
+		return false, err
+	}
+
+	s := sling.New().Set(VaultToken, t.Token)
+
+	req, err := s.New().Base(secretBaseURL).Get(credPath).Request()
+	resp, err := c.Do(req)
+	if err != nil {
+		errStr := fmt.Sprintf("Failed to retrieve credentials with path as %s with error %s", credPath, err.Error())
+		return false, errors.New(errStr)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		lc.Info(fmt.Sprintf("No credential path found, setting up the credentials for %s", credPath))
+		return false, nil
+	}
+
+	lc.Info(fmt.Sprintf("%s - %d", credPath, resp.StatusCode))
+
+	var result map[string]interface{}
+
+	by, _ := ioutil.ReadAll(resp.Body)
+
+	json.Unmarshal(by, &result)
+
+	credentials := result["data"].(map[string]interface{})
+
+	if len(credentials) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func initMongoDBCredentials(config *tomlConfig, secretBaseURL string, c *http.Client) error {
+
+	//TODO: we only covert coredata credential for this release, the rest needs to be implemented later.
```

Comment:
Wait, what? The churn to get this feature in before the code freeze isn't worth it for one service. @jpwhitemn

Reply:
It was discussed and decided in the previous meeting that we will start with coredata service only for this release so that we will know what issue we may encounter before applying it to all the core microservice.

Reply:
FWIW I don't recall this decision being made but I also may have missed that WG meeting

Reply:
this file has been removed and the feature will be implemented in a separated PR.